chia/consensus/blockchain.py

```diff
@@ -678,11 +674,11 @@ def block_record(self, header_hash: bytes32) -> BlockRecord:
         return self.__block_records[header_hash]
 
     def height_to_block_record(self, height: uint32) -> BlockRecord:
-        header_hash = self.height_to_hash(height)
-        # TODO: address hint error and remove ignore
-        #       error: Argument 1 to "block_record" of "Blockchain" has incompatible type "Optional[bytes32]"; expected
-        #       "bytes32"  [arg-type]
-        return self.block_record(header_hash)  # type: ignore[arg-type]
+        # Precondition: height is in the blockchain
+        header_hash: Optional[bytes32] = self.height_to_hash(height)
```

Comment:
Looks like about 33 failure-of-some-form-or-api-return-failures vs. 8 keep-running-in-some-way, as a quick rough estimation searching for all uses of `.height_to_hash()` across the multiple classes that have it (all under `BlockchainInterface`'s guidance I think).  Aside from that I just kind of like to make sure that the keep-running cases have been thought through well and explicitly addressed rather than just happening accidentally because a dev forgot to be really careful about the `None` case.  Also, for APIs, centralizing failure handling seems nice, though there's some complexity with 'just fail' vs. 'fail and ban for awhile' and such.